app/Models/User.php

```diff
@@ -681,13 +681,10 @@ public function setCollegist($objectName): void
      */
     public function isResident(): bool
     {
-        if($this->verified == false) {
-            return $this->roles()
-            ->where('role_id', Role::collegist()->id)
-            ->where('object_id', RoleObject::firstWhere('name', Role::RESIDENT)->id)
-            ->exists();
-        }
-        return $this->hasRole([Role::COLLEGIST => Role::RESIDENT]);
+        return $this->roles()
+        ->where('role_id', Role::collegist()->id)
+        ->where('object_id', RoleObject::firstWhere('name', Role::RESIDENT)->id)
+        ->exists();
```

Comment:
Just comment the previous one out and add a todo. 
We need to remove the first branch when we detach roles from the applicants so it should just be  $this->hasRole(...).

For the bug you see, idk, we should find it asap, but if you can not you can leave this like that.

Reply:
Oh; it was the cache:D After running `php artisan optimize:clear`, it was fine. I revert it.